### ISSUE TITLE
oc new-app <repo> --strategy=docker constructs bad app 

### ISSUE THREAD

**Issue by rhcarvalho** [openshift/origin#6347]
tl;dr: `new-app` should not come up with a builder image for the app and reference it in the build config strategy when strategy=docker.
---

Use case: I have a Dockerfile in my source repository and want to setup a build that will simply build that Dockerfile.

"Go-to" solution:

``` console
$ oc new-app https://github.com/openshift/ruby-hello-world --strategy=docker
```

Problem: `new-app` sets up a BuildConfig that will replace the FROM line in my Dockerfile with a builder image (the image `openshift/ruby-22-centos7` in the example above).

YAML output of `new-app`:

``` yaml
# ...
- apiVersion: v1
  kind: BuildConfig
  metadata:
    annotations:
      openshift.io/generated-by: OpenShiftNewApp
    labels:
      app: ruby-hello-world
    name: ruby-hello-world
  spec:
    output:
      to:
        kind: ImageStreamTag
        name: ruby-hello-world:latest
    source:
      git:
        uri: https://github.com/openshift/ruby-hello-world
      type: Git
    strategy:
      dockerStrategy:
        from: # <-- this will replace the FROM line of the Dockerfile in the source repository
          kind: ImageStreamTag
          name: ruby-22-centos7:latest
      type: Docker
# ...
```

`new-app` should not come up with a builder image for the app and reference it in the build config strategy when strategy=docker.

The strategy should be:

``` yaml
   strategy:
      dockerStrategy: {}
```

**Comment by rhcarvalho** [comment]
cc @php-coder

**Comment by bparees** [comment]
debatable since someone could explicitly define an image they want, eg:
oc new-app some/image~https://github.com/some/repo.git

at which point they are explicitly saying they do want their Dockerfile updated w/ the value of some/image.

so we'd only want to do what you propose if they didn't explicitly provide an image to be used with the repo.  which is more fun special casing for new-app.

**Comment by bparees** [comment]
Having had some more time to mull this over, i'm not convinced the current behavior is wrong.  New-app is designed around the openshift experience, which includes setting up imagestreams for your docker images and having your builds reference imagestreams/be triggered by imagestream changes.  

What you are describing sounds like an advanced use case at which point a user can setup their own buildconfig however they want.  I don't like the idea of providing different experiences based on whether your input included a dockerfile or not.

It's also a pretty big change in new-app behavior that isn't backwards compatible to the current behavior.

**Comment by bparees** [comment]
(a few more thoughts on this:   I think it will only lead to more confusion on the part of users as to why sometimes they get an imagestream and other times they do not.  new-app has enough "magic" behavior and i'd rather not add more.  Also in the year since the original issue was opened, no one else has raised a concern about the current behavior.)
